// Copyright 2016 Rodrigo Jose Hernandez Cordoba
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";
package AeonGames;

import "vector2.proto";
import "vector3.proto";
import "vector4.proto";

message PropertyValueBuffer
{
	string name = 1;
	oneof value
	{
		float    scalar_float = 2;
		Vector2Buffer vector2 = 3;
		Vector3Buffer vector3 = 4;
		Vector4Buffer vector4 = 5;
		string        texture = 6;
	}
}

message MaterialBuffer
{
	repeated PropertyValueBuffer property_values = 1;
}
